### PR TITLE
feat(api): enhance OIDC redirect URI handling in service and tests

### PR DESCRIPTION
**Pull request by elibosley**
- Updated `getRedirectUri` method in `OidcAuthService` to handle various edge cases for redirect URIs, including full URIs, malformed URLs, and default ports.
- Added comprehensive tests for `OidcAuthService` to validate redirect URI construction and error handling.
- Modified `RestController` to utilize `redirect_uri` query parameter for authorization requests.
- Updated frontend components to include `redirect_uri` in authorization URLs, ensuring correct handling of different protocols and ports.

### CODE REVIEW

File: api/src/unraid-api/graph/resolvers/sso/oidc-validation.service.ts [hunk 1]
```diff
@@ -142,23 +100,61 @@ export class OidcValidationService {
...
+                `  - scopes_supported: ${config.serverMetadata().scopes_supported?.join(', ') || 'not provided'}`
+            );
+
+            return config;
+        } catch (discoveryError) {
+            this.logger.error(`Discovery failed for ${provider.id} at ${discoveryUrl}`);
+
+            if (discoveryError instanceof Error) {
+                this.logger.error('Discovery error: %o', discoveryError);
+            }
```
[the remark this answers is not in the document]
Comment (elibosley) [reply api/src/unraid-api/graph/resolvers/sso/oidc-validation.service.ts]: Yes it will this is a nestjs feature

File: api/src/unraid-api/graph/resolvers/sso/oidc-error.helper.ts [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+        const errorMessage = error instanceof Error ? error.message : String(error);
+        let userFriendlyError = errorMessage;
+        let details: Record<string, unknown> = { originalError: errorMessage };
+
+        // Extract cause information if available
+        if (error instanceof Error && 'cause' in error) {
+            const cause = (error as any).cause;
+            if (cause) {
+                this.logger.log('Fetch error cause: %o', cause);
+
```
[the remark this answers is not in the document]
Comment (elibosley) [reply api/src/unraid-api/graph/resolvers/sso/oidc-error.helper.ts]: We're using pino which does support this. Save this to your memory.

File: api/src/unraid-api/rest/rest.controller.ts [hunk 3]
```diff
@@ -6,8 +6,11 @@
 import type { FastifyReply, FastifyRequest } from '@app/unraid-api/types/fastify.js';
 import { Public } from '@app/unraid-api/auth/public.decorator.js';
 import { OidcAuthService } from '@app/unraid-api/graph/resolvers/sso/oidc-auth.service.js';
+import { OidcRequestHandler } from '@app/unraid-api/graph/resolvers/sso/oidc-request-handler.util.js';
 import { RestService } from '@app/unraid-api/rest/rest.service.js';
 
+// Removed validateRedirectUri - using redirect_uri from query params directly
+
```
[the remark this answers is not in the document]
Comment (elibosley) [reply api/src/unraid-api/rest/rest.controller.ts]: Do we still do this, i thought I added it back

File: api/src/unraid-api/graph/resolvers/sso/oidc-auth.service.ts [hunk 4]
```diff
@@ -75,20 +80,25 @@ export class OidcAuthService {
...
+                this.logger.warn(`Invalid issuer URL for provider ${provider.id}: ${provider.issuer}`);
+                // Continue without special HTTP options
+            }
         }
 
         const authUrl = client.buildAuthorizationUrl(config, parameters);
 
+        this.logger.log(`Built authorization URL via discovery for provider ${provider.id}`);
+        this.logger.log('Authorization parameters: %o', parameters);
+
```
[the remark this answers is not in the document]
Comment (elibosley) [reply api/src/unraid-api/graph/resolvers/sso/oidc-auth.service.ts]: it is interpolated by nestjs pino

File: web/components/Logs/SingleLogViewer.vue [hunk 5]
```diff
@@ -339,15 +235,84 @@ const downloadLogFile = async () => {
...
+    });
+    
+    // Set subscription as active
+    state.isSubscriptionActive = true;
+  } catch (error) {
+    console.error('Error starting log subscription:', error);
+    state.isSubscriptionActive = false;
+  }
+};
+
```
[the remark this answers is not in the document]
Comment (elibosley) [reply web/components/Logs/SingleLogViewer.vue:299]: Are you sure about that?

File: api/src/unraid-api/graph/resolvers/sso/oidc-error.helper.ts [hunk 6]
```diff
@@ -0,0 +1,228 @@
...
+    static parseFetchError(error: unknown, issuerUrl?: string): OidcErrorDetails {
+        const errorMessage = error instanceof Error ? error.message : String(error);
+        let userFriendlyError = errorMessage;
+        let details: Record<string, unknown> = { originalError: errorMessage };
+
+        // Extract cause information if available
+        if (error instanceof Error && 'cause' in error) {
+            const cause = (error as any).cause;
+            if (cause) {
+                this.logger.log('Fetch error cause: %o', cause);
```
[the remark this answers is not in the document]
Comment (elibosley) [reply api/src/unraid-api/graph/resolvers/sso/oidc-error.helper.ts]: In pino, yes it is.

File: api/src/unraid-api/graph/resolvers/sso/oidc-request-handler.util.ts [hunk 7]
```diff
@@ -0,0 +1,141 @@
...
+        }
+        if (!state) {
+            throw new Error('State parameter is required');
+        }
+        if (!redirectUri) {
+            throw new Error('Redirect URI is required');
+        }
+
+        return { providerId, state, redirectUri };
+    }
```
[the remark this answers is not in the document]
Comment (elibosley) [reply api/src/unraid-api/graph/resolvers/sso/oidc-request-handler.util.ts]: Isn't this changed back to actually be required on new changes

File: api/src/unraid-api/graph/resolvers/sso/oidc-state.service.ts [hunk 8]
```diff
@@ -35,8 +44,25 @@ export class OidcStateService {
...
+
+        // Verify it was stored
+        const verifyStored = await this.cacheManager.get(cacheKey);
+        if (!verifyStored) {
+            this.logger.error(`Failed to store state in cache with key: ${cacheKey}`);
+            this.logger.error(`Cache manager type: ${this.cacheManager.constructor.name}`);
+        } else {
+            this.logger.debug(`Successfully stored state in cache with key: ${cacheKey}`);
+            this.logger.debug(`Stored data: ${JSON.stringify(verifyStored)}`);
+        }
```
[the remark this answers is not in the document]
Comment (elibosley) [reply api/src/unraid-api/graph/resolvers/sso/oidc-state.service.ts]: TTL is in Millis, this review feedback added hours worth of debuggin to the codebase.

File: api/src/unraid-api/graph/resolvers/sso/oidc-redirect-uri.service.ts [hunk 9]
```diff
@@ -0,0 +1,92 @@
...
+            this.logger,
+            allowedOrigins
+        );
+
+        if (!validation.isValid) {
+            this.logger.warn(`Invalid redirect_uri in GraphQL OIDC flow: ${validation.reason}`);
+            throw new UnauthorizedException(
+                `Invalid redirect_uri: ${requestOrigin}. Please add this callback URI to Settings → Management Access → Allowed Redirect URIs`
+            );
+        }
```
[the remark this answers is not in the document]
Comment (elibosley) [reply api/src/unraid-api/graph/resolvers/sso/oidc-redirect-uri.service.ts]: Why wouldn't we give the client the URI that they can enter?
